tests/Aql/DistinctCollectExecutorTest.cpp

```diff
@@ -50,310 +51,72 @@ namespace arangodb {
-                                     regToKeep, std::move(readableInputRegisters),
-                                     std::move(writeableOutputRegisters),
-                                     std::move(groupRegisters), trx);
-  block.reset(new AqlItemBlock(itemBlockManager, 1000, 2));
-
-  SingleRowFetcherHelper<::arangodb::aql::BlockPassthrough::Disable> fetcher(itemBlockManager, input.steal(), true);
-  DistinctCollectExecutor testee(fetcher, infos);
-
-  OutputAqlItemRow result(std::move(block), infos.getOutputRegisters(),
-                          infos.registersToKeep(), infos.registersToClear());
-  std::tie(state, stats) = testee.produceRows(result);
-  ASSERT_EQ(state, ExecutionState::WAITING);
-  ASSERT_FALSE(result.produced());
-
-  std::tie(state, stats) = testee.produceRows(result);
-  ASSERT_EQ(state, ExecutionState::DONE);
-  ASSERT_FALSE(result.produced());
-}
-
-TEST_F(DistinctCollectExecutorTest,
-       there_are_rows_in_the_upstream_no_distinct_values_the_producer_doesnt_wait) {
-  groupRegisters.emplace_back(std::make_pair<RegisterId, RegisterId>(1, 0));
-  readableInputRegisters.insert(0);
-  writeableOutputRegisters.insert(1);
-  RegisterId nrOutputRegister = 2;
-  DistinctCollectExecutorInfos infos(1 /*nrInputReg*/,
-                                     nrOutputRegister /*nrOutputReg*/, regToClear,
-                                     regToKeep, std::move(readableInputRegisters),
-                                     std::move(writeableOutputRegisters),
-                                     std::move(groupRegisters), trx);
-  block.reset(new AqlItemBlock(itemBlockManager, 1000, nrOutputRegister));
-
-  auto input = VPackParser::fromJson("[ [1], [2] ]");
-  SingleRowFetcherHelper<::arangodb::aql::BlockPassthrough::Disable> fetcher(itemBlockManager, input->steal(), false);
-  DistinctCollectExecutor testee(fetcher, infos);
-
-  OutputAqlItemRow result(std::move(block), infos.getOutputRegisters(),
-                          infos.registersToKeep(), infos.registersToClear());
-
-  std::tie(state, stats) = testee.produceRows(result);
-  ASSERT_EQ(state, ExecutionState::HASMORE);
-  ASSERT_TRUE(result.produced());
-  result.advanceRow();
-
-  std::tie(state, stats) = testee.produceRows(result);
-  ASSERT_EQ(state, ExecutionState::DONE);
-  ASSERT_TRUE(result.produced());
-  result.advanceRow();
-
-  std::tie(state, stats) = testee.produceRows(result);
-  ASSERT_EQ(state, ExecutionState::DONE);
-  ASSERT_FALSE(result.produced());
-
-  auto block = result.stealBlock();
-  AqlValue x = block->getValue(0, 1);
-  ASSERT_TRUE(x.isNumber());
-  ASSERT_EQ(x.toInt64(), 1);
-
-  AqlValue z = block->getValue(1, 1);
-  ASSERT_TRUE(z.isNumber());
-  ASSERT_EQ(z.toInt64(), 2);
+TEST_P(DistinctCollectExecutorTest, split_1) {
+  auto [split] = GetParam();
+
+  ExecutorTestHelper(*fakedQuery)
+      .setExecBlock<DistinctCollectExecutor>(std::move(infos))
+      .setInputValueList(1, 1, 1, 2, 3, 4, 4, 5)
```

Comment:
Can you either add a test where the input is not ordered, or permutate over the order of inputs?

Returning sorted input in a distinct way is easier then unsorted input.